### PR TITLE
Add Iron160 I160-S and I160-H PCBs support

### PR DESCRIPTION
**Pull request by Gondolindrim**
Adds support for Iron160 PCBs in both variants: hotswap and solderable.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/smithrune/iron160/readme.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Iron 160 QMK folder
+
+![Iron160](https://i.imgur.com/8B2l2pn.jpg)
```
Comment (drashna) [inline keyboards/smithrune/iron160/readme.md:3]:
```suggestion
![Iron160](https://i.imgur.com/8B2l2pnh.jpg)
```

File: keyboards/smithrune/iron160/iron160_s/iron160_s.c [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "iron160_s.h"
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) writePin(LED_CAPS_LOCK_PIN, led_state.caps_lock);
+    return res;
+}
```
Comment (drashna) [inline keyboards/smithrune/iron160/iron160_s/iron160_s.c:24]: If the pin is set, this shouldn't be needed, at all. 

So the entire file could go, actually.

File: keyboards/smithrune/iron160/iron160_s/config.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 14
```
Comment (drashna) [inline keyboards/smithrune/iron160/iron160_s/config.h:22]:
```suggestion
```

File: keyboards/smithrune/iron160/iron160_h/config.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+/* key matrix size */
+#define MATRIX_ROWS 5
+#define MATRIX_COLS 14
```
Comment (drashna) [inline keyboards/smithrune/iron160/iron160_h/config.h:22]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/smithrune/iron160/iron160_h/config.h (hunk 4 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron160/iron160_h/config.h]: Removed in [0fa8847](https://github.com/qmk/qmk_firmware/pull/20077/commits/0fa884797f8e940a19913ec6a2f257d1e8877c77)

File: keyboards/smithrune/iron160/iron160_s/config.h (hunk 3 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron160/iron160_s/config.h]: Removed in [0fa8847](https://github.com/qmk/qmk_firmware/pull/20077/commits/0fa884797f8e940a19913ec6a2f257d1e8877c77)

File: keyboards/smithrune/iron160/readme.md (hunk 1 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron160/readme.md]: Added in 
d7770bf

File: keyboards/smithrune/iron160/iron160_s/config.h [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+*/
+
+#pragma once
+
+//                        C0 , C1 , C2 , C3 , C4 , C5 , C6 , C7 , C8 , C9 , C10, C11, C12, C13 
+#define MATRIX_COL_PINS { A3 , B9 , B8 , B5 , B4 , B3 , A6 , A5 , A4 , A2 , A1 , A0 , C15, C14}
+//                        R0 , R1 , R2 , R3 , R4
+#define MATRIX_ROW_PINS { A10, A15, C13, B1 , B10}
+#define DIODE_DIRECTION COL2ROW
+
```
Comment (fauxpark) [inline keyboards/smithrune/iron160/iron160_s/config.h:25]:
```suggestion
```

File: keyboards/smithrune/iron160/iron160_s/config.h [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+#define BACKLIGHT_PIN           A7
+#define BACKLIGHT_PWM_DRIVER    PWMD3
+#define BACKLIGHT_PWM_CHANNEL   2
+#define BACKLIGHT_PAL_MODE      1
+#define BACKLIGHT_LEVELS 20
+#define BACKLIGHT_BREATHING
+#define BREATHING_PERIOD 3
+#define BACKLIGHT_ON_STATE 1
```
Comment (fauxpark) [inline keyboards/smithrune/iron160/iron160_s/config.h:33]:
```suggestion
#define BACKLIGHT_PWM_DRIVER    PWMD3
#define BACKLIGHT_PWM_CHANNEL   2
#define BACKLIGHT_PAL_MODE      1
```

File: keyboards/smithrune/iron160/iron160_s/info.json [hunk 7]
```diff
@@ -0,0 +1,82 @@
+{
+    "keyboard_name": "I160-S",
+    "manufacturer": "SmithRune",
+    "url": "",
+    "maintainer": "Gondolindrim",
+    "usb": {
+        "vid": "0x8384",
+        "pid": "0x1653",
+        "device_version": "0.0.1"
+    },
```
Comment (fauxpark) [inline keyboards/smithrune/iron160/iron160_s/info.json:10]:
```suggestion
    },
    "processor": "STM32F072",
    "bootloader: "stm32-dfu",
    "matrix_pins": {
        "cols": ["A3", "B9", "B8", "B5", "B4", "B3", "A6", "A5", "A4", "A2", "A1", "A0", "C15", "C14"],
        "rows": ["A10", "A15", "C13", "B1", "B10"]
    },
    "diode_direction": "COL2ROW",
    "backlight": {
        "pin": "A7",
        "levels": 20,
        "breathing": true,
        "breathing_period": 3
    },
    "indicators": {
        "caps_lock": "B0"
    },
```

File: keyboards/smithrune/iron160/iron160_s/config.h [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+#define LED_CAPS_LOCK_PIN B0
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE    5
+
```
Comment (fauxpark) [inline keyboards/smithrune/iron160/iron160_s/config.h:37]:
```suggestion
```

File: keyboards/smithrune/iron160/iron160_s/iron160_s.h [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#define ___ KC_NO
+
+#include "quantum.h"
+
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/smithrune/iron160/iron160_s/iron160_s.h:24]: Layout can be moved to info.json, and these two files can be removed.

File: keyboards/smithrune/iron160/iron160_h/rules.mk [hunk 10]
```diff
@@ -0,0 +1,26 @@
+# MCU name
+MCU = STM32F072
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
```
Comment (fauxpark) [inline keyboards/smithrune/iron160/iron160_h/rules.mk:6]:
```suggestion
```

File: keyboards/smithrune/iron160/iron160_h/info.json [hunk 11]
```diff
@@ -0,0 +1,79 @@
+{
+    "keyboard_name": "I160-H",
+    "manufacturer": "SmithRune",
+    "url": "",
+    "maintainer": "Gondolindrim",
+    "usb": {
+        "vid": "0x8384",
+        "pid": "0x1648",
+        "device_version": "0.0.1"
+    },
```
Comment (fauxpark) [inline keyboards/smithrune/iron160/iron160_h/info.json:10]:
```suggestion
    },
    "processor": "STM32F072",
    "bootloader": "stm32-dfu",
    "matrix_pins": {
        "cols": ["A3", "B9", "B8", "B5", "B4", "B3", "A6", "A5", "A4", "A2", "A1", "A0", "C15", "C14"],
        "rows": ["A10", "A15", "C13", "A7", "B0"]
    },
    "diode_direction": "COL2ROW",
```

File: keyboards/smithrune/iron160/iron160_h/config.h [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+//                        R0 , R1 , R2 , R3 , R4
+#define MATRIX_ROW_PINS { A10, A15, C13, A7 , B0 }
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN B15
+#define RGBLED_NUM 1
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE    5
+
```
Comment (fauxpark) [inline keyboards/smithrune/iron160/iron160_h/config.h:31]:
```suggestion
#define RGB_DI_PIN B15
#define RGBLED_NUM 1

```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/smithrune/iron160/iron160_h/info.json [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+   "processor": "STM32F072",
+   "bootloader": "stm32-dfu",
+   "matrix_pins": {
+       "cols": ["A3" , "B9" , "B8" , "B5" , "B4", "B3" , "A6" , "A5" , "A4" , "A2" , "A1" , "A0" , "C15" , "C14"],
+       "rows": ["A10", "A15", "C13", "A7" , "B0"]
+   },
```
Comment (waffle87) [inline keyboards/smithrune/iron160/iron160_h/info.json:16]:
```suggestion
   },
   "rgblight": {
       "led_count": 1,
       "pin": "B15"
   },
```

File: keyboards/smithrune/iron160/iron160_h/config.h [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+#define RGB_DI_PIN B15
+#define RGBLED_NUM 1
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+/* Defining RGB default brightness to zero */
+#define RGBLIGHT_DEFAULT_VAL 0
```
Comment (waffle87) [inline keyboards/smithrune/iron160/iron160_h/config.h:29]: This file can be removed with the above suggestion. (settings RGB light brightness to 0 seems like it'll be confusing for users)

File: keyboards/smithrune/iron160/iron160_h/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______
+    ),
+    [3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______
+    ),
```
Comment (waffle87) [inline keyboards/smithrune/iron160/iron160_h/keymaps/default/keymap.c:48]:
```suggestion
        QK_BOOT  , _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
        _______, _______, _______,                            _______,                            _______, _______, _______, _______
    )
```

File: keyboards/smithrune/iron160/iron160_s/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                   _______, _______, _______, _______, _______
+    ),
+    [3] = LAYOUT(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+        _______, _______, _______,                            _______,                   _______, _______, _______, _______, _______
+    ),
```
Comment (waffle87) [inline keyboards/smithrune/iron160/iron160_s/keymaps/default/keymap.c:48]:
```suggestion
    )
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/smithrune/iron160/iron160_h/config.h (hunk 14 above)
Comment (Gondolindrim) [reply keyboards/smithrune/iron160/iron160_h/config.h:29]: This is intentional -- we want the RGBs turned off by default.

**Review by waffle87 — APPROVED** [review]

**Pull request merged**